source/PlayerInfo.cpp

```diff
@@ -2579,3 +2621,46 @@ bool PlayerInfo::CanBeSaved() const
 {
 	return (!isDead && planet && system && !firstName.empty() && !lastName.empty());
 }
+
+
+
+// Called by conversation callbacks to destroy or capture the player's flagship.
+// Either method results in the death of the player.
+void PlayerInfo::Die(int response)
+{
+	if(GetPlanet() || !flagship)
+		Die(true);
+	else
+	{
+		Die();
+		auto it = find(ships.begin(), ships.end(), flagship);
+		if(response == Conversation::EXPLODE)
+			flagship->Destroy();
+		else
+		{
+			flagship->SetIsYours(false);
```

Comment:
Technically speaking, the flagship's crew should be reduced by 1 (since you're dead). But, that would create an issue if you're boarded by someone with only 1 crew, and you only have 1 crew too, because there would not be enough crew to control both ships.

Reply:
I thought of reducing it, but since the player is dead and can no longer inspect the ship that just boarded them, it was better to keep as-is than reduce it and perhaps cause an issue.